tests/h/views/activity_test.py

```diff
@@ -117,17 +117,17 @@ class TestGroupSearchController(object):
 
     """Tests unique to GroupSearchController."""
 
-    def test_renders_join_template_when_no_read_permission(self, controller, pyramid_request, group):
+    def test_renders_join_template_when_no_read_permission(self, controller, pyramid_request, group, factories):
         """When the request has no read permission but join, it should render the join template."""
 
         def fake_has_permission(permission, context=None):
             if permission == 'read':
                 return False
-            if permission == 'join':
+            elif permission == 'join':
                 return True
         pyramid_request.has_permission = mock.Mock(side_effect=fake_has_permission)
         pyramid_request.override_renderer = mock.PropertyMock()
-
+        pyramid_request.user = factories.User()
```

Comment:
One way to improve these tests (can be done in a separate pr) might be to add separate fixtures for `not_logged_in_request`, `logged_in_but_not_a_member_request`, `group_member_request`, etc. Then each test just needs to use the fixture it wants.